### PR TITLE
Implemented basic serialization (json)

### PR DESCRIPTION
**Pull request by gi0baro**
@mdipierro Please don't merge this yet, needs some cleaning up.

### CODE REVIEW

**Comment by gi0baro** [comment]
@mdipierro @niphlod @ilvalle what do you think about this?
It will need a small update on web2py, since `DAL.serializers` is now expected to be a `dict` object.

File: pydal/helpers/classes.py [hunk 1]
```diff
@@ -295,3 +299,16 @@ def file_delete(self,filename):
         self.db.executesql(query)
         self.db.commit()
 
+
+class Serializable(object):
+    def as_dict(self):
```
Comment (ilvalle) [inline pydal/helpers/classes.py]: isn't better to write ```as_dict(self, flat=False, sanitize=True)``` ?

**Comment by ilvalle** [comment]
This PR removes some redundant code, which is good.
I'm wondering if DAL can inherit from Serializable ?

(try to rebase commits, you can do that even if there is the PR open)

**Comment by gi0baro** [comment]
@ilvalle good point about the `as_dict` method, I've just updated it.

Regarding the `DAL` inheritance from `Serializable` I'm quite worried about the "pickle serialization" issues, but I can try.

**Comment by gi0baro** [comment]
I've added inheritance for `DAL` from `Serializable`. Pickling seems to work (only tested locally, not inside web2py)

**Comment by ilvalle** [comment]
I've the feeling that the tests we have are not enough to check all your contributions

File: pydal/helpers/serializers.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        value = jsonlib.dumps(value, default=self._json_parse)
+        return value.replace(ur'\u2028', '\\u2028').replace(
+            ur'\2029', '\\u2029')
+
+    def yaml(self, value):
+        if self._custom_.get('yaml') is not None:
+            return self._custom_.get('yaml')(value)
+        try:
+            from yaml import dump
+        except:
```
Comment (mdipierro) [inline pydal/helpers/serializers.py]: except ImportError?

Comment (gi0baro) [inline pydal/helpers/serializers.py]: Right, I will update the PR.

Comment (mdipierro) [inline pydal/helpers/serializers.py]: How about at the top of the file?

     try:
           from yaml import dump as yaml_dump
           HAVE_YAML = true
     except ImportError
           HAVE_YAML = False

**Comment by gi0baro** [comment]
I've just updated the yaml import try/except block.

**Comment by gi0baro** [comment]
I think this is ready for merge.

**Comment by ilvalle** [comment]
Can you rebase the PR in a few commits? 
If rebase is not possible, I usually do http://stackoverflow.com/questions/3697178/git-merge-all-changes-from-another-branch-as-a-single-commit

**Comment by gi0baro** [comment]
I've just rebased the PR with 4 commits. @mdipierro please review and merge it

**Comment by mdipierro** [comment]
Traceback (most recent call last):
File "/Users/massimodipierro/Dropbox/web2py/gluon/restricted.py", line 227, in restricted
exec ccode in environment
File "/Users/massimodipierro/Dropbox/web2py/applications/CAAForecasting/models/db.py", line 20, in 
db = DAL(myconf.take('db.uri'), pool_size=myconf.take('db.pool_size', cast=int), check_reserved=['all'])
File "/Users/massimodipierro/Dropbox/web2py/gluon/packages/dal/pydal/base.py", line 171, in call
obj = super(MetaDAL, cls).call(args, *kwargs)
File "/Users/massimodipierro/Dropbox/web2py/gluon/packages/dal/pydal/base.py", line 475, in init
for k, v in self.serializers.items():
AttributeError: 'module' object has no attribute 'items'

**Comment by gi0baro** [comment]
Yes, as I wrote, we need to change the way serializers are passed to pyDAL. This is not pyDAL related but about web2py, I will post a PR soon.

**Pull request merged**